healthcheck/client.go

```diff
@@ -0,0 +1,28 @@
+package healthcheck
+
+import (
+	rchttp "github.com/ONSdigital/dp-rchttp"
+	"sync"
+)
+
+type Client struct {
+	Clienter   rchttp.Clienter
+	Check      *Check
+	Checker    *Checker
+	mutex *sync.Mutex
+}
+
+// NewClient returns a pointer to a new instantiated Client with
+// the provided checker function and an optional rchttp.Clienter
+func NewClient(cli rchttp.Clienter, checker *Checker) *Client {
+	if cli == nil {
```

Comment:
Is this check needed? I wonder if it would possibly be better to do
```
if cli == nil {
    return errors.new("expected clienter but non provided")
}
```
That way the onus is clearly on the caller to provide a valid `clienter`. It avoids questions like: _which clienter am I using?_ when debugging issues.